--- origsrc/zathura-cb-0.1.1/cb.c	2014-03-26 13:10:12.163878700 -0500
+++ src/zathura-cb-0.1.1/cb.c	2014-03-26 13:10:05.494531800 -0500
@@ -122,11 +122,11 @@ read_archive(cb_document_t* cb_document,
     return false;
   }
 
-  archive_read_support_filter_all(a);
+  archive_read_support_compression_all(a);
   archive_read_support_format_all(a);
   int r = archive_read_open_filename(a, archive, (size_t) LIBARCHIVE_BUFFER_SIZE);
   if (r != ARCHIVE_OK) {
-    archive_read_free(a);
+    archive_read_finish(a);
     return false;
   }
 
@@ -135,7 +135,7 @@ read_archive(cb_document_t* cb_document,
     if (r < ARCHIVE_WARN) {
       // let's ignore warnings ... they are non-fatal errors
       archive_read_close(a);
-      archive_read_free(a);
+      archive_read_finish(a);
       return false;
     }
 
@@ -156,7 +156,7 @@ read_archive(cb_document_t* cb_document,
   }
 
   archive_read_close(a);
-  archive_read_free(a);
+  archive_read_finish(a);
   return true;
 }
 
@@ -272,7 +272,7 @@ load_pixbuf_from_archive(const char* arc
     return NULL;
   }
 
-  archive_read_support_filter_all(a);
+  archive_read_support_compression_all(a);
   archive_read_support_format_all(a);
   int r = archive_read_open_filename(a, archive, LIBARCHIVE_BUFFER_SIZE);
   if (r != ARCHIVE_OK) {
@@ -283,7 +283,7 @@ load_pixbuf_from_archive(const char* arc
   while ((r = archive_read_next_header(a, &entry)) != ARCHIVE_EOF) {
     if (r < ARCHIVE_WARN) {
       archive_read_close(a);
-      archive_read_free(a);
+      archive_read_finish(a);
       return NULL;
     }
 
@@ -294,7 +294,7 @@ load_pixbuf_from_archive(const char* arc
     GInputStream* is = g_memory_input_stream_new();
     if (is == NULL) {
       archive_read_close(a);
-      archive_read_free(a);
+      archive_read_finish(a);
       return NULL;
     }
     GMemoryInputStream* mis = G_MEMORY_INPUT_STREAM(is);
@@ -305,7 +305,7 @@ load_pixbuf_from_archive(const char* arc
     while ((r = archive_read_data_block(a, &buf, &size, &offset)) != ARCHIVE_EOF) {
       if (r < ARCHIVE_WARN) {
         archive_read_close(a);
-        archive_read_free(a);
+        archive_read_finish(a);
         g_object_unref(mis);
         return NULL;
       }
@@ -313,7 +313,7 @@ load_pixbuf_from_archive(const char* arc
       void* tmp = g_malloc0(size);
       if (tmp == NULL) {
         archive_read_close(a);
-        archive_read_free(a);
+        archive_read_finish(a);
         g_object_unref(mis);
         return NULL;
       }
@@ -325,19 +325,19 @@ load_pixbuf_from_archive(const char* arc
     GdkPixbuf* pixbuf = gdk_pixbuf_new_from_stream(is, NULL, NULL);
     if (pixbuf == NULL) {
       archive_read_close(a);
-      archive_read_free(a);
+      archive_read_finish(a);
       g_object_unref(mis);
       return NULL;
     }
 
     archive_read_close(a);
-    archive_read_free(a);
+    archive_read_finish(a);
     g_object_unref(mis);
     return pixbuf;
   }
 
   archive_read_close(a);
-  archive_read_free(a);
+  archive_read_finish(a);
   return NULL;
 }
 
